### PR TITLE
Test Cases for Gotrack2 Blob SDK

### PR DESCRIPTION
**Pull request by mohsha-msft**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] Apache v2 license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azblob/zc_appendBlobClient.go [hunk 1]
```diff
@@ -25,36 +25,71 @@ func NewAppendBlobClient(blobURL string, cred azcore.Credential, options *connec
...
-		},
-		u: ab.u,
+		client:     &appendBlobClient{con: con},
+		u:          ab.u,
+		BlobClient: BlobClient{client: &blobClient{con: con}},
 	}
 }
 
+func (ab AppendBlobClient) GetAccountInfo(ctx context.Context) (BlobGetAccountInfoResponse, error) {
+	blobClient := BlobClient{client: &blobClient{ab.client.con, nil}}
```
Comment (zezha-msft) [inline sdk/storage/azblob/zc_appendBlobClient.go:77]: We shouldn't need to create a blobClient, refer to track 1.5 where we call the parent directly. But actually these methods don't have to be created on the children, right? They are all inherited.

File: sdk/storage/azblob/zc_containerClient.go [hunk 2]
```diff
@@ -145,17 +148,19 @@ func (c ContainerClient) GetProperties(ctx context.Context, gpo *GetPropertiesOp
...
-	metadataOptions := ContainerSetMetadataOptions{
-		Metadata: &metadata,
-	}
-
-	mac, lac := ac.pointers()
-
-	return c.client.SetMetadata(ctx, &metadataOptions, lac, mac)
+func (c ContainerClient) SetMetadata(ctx context.Context, options *SetMetadataContainerOptions) (ContainerSetMetadataResponse, error) {
+	// TODO: Ask Ze/Adele: Why we introduced this check. We should let service do this kind of validations.
+	//if ac != nil && ac.ModifiedAccessConditions != nil {
```
Comment (zezha-msft) [inline sdk/storage/azblob/zc_containerClient.go:153]: The access conditions are totally ignored, so the cx may be getting the undesired experience.

**Review by zezha-msft — APPROVED** [review]
Good work! Some minor comments.

File: sdk/storage/azblob/zc_appendBlobClient.go [hunk 3]
```diff
@@ -25,36 +25,71 @@ func NewAppendBlobClient(blobURL string, cred azcore.Credential, options *connec
...
+// WithSnapshot creates a new AppendBlobURL object identical to the source but with the specified snapshot timestamp.
+// Pass "" to remove the snapshot returning a URL to the base blob.
 func (ab AppendBlobClient) WithSnapshot(snapshot string) AppendBlobClient {
 	p := NewBlobURLParts(ab.URL())
 	p.Snapshot = snapshot
 	snapshotURL := p.URL()
+	con := newConnectionWithPipeline(snapshotURL.String(), ab.client.con.p)
+	return AppendBlobClient{
+		client:     &appendBlobClient{con: con},
+		u:          ab.u,
```
[the remark this answers is not in the document]
Comment (mohsha-msft) [reply sdk/storage/azblob/zc_appendBlobClient.go:57]: Yes correct

File: sdk/storage/azblob/zc_appendBlobClient.go (hunk 1 above)
Comment (mohsha-msft) [reply sdk/storage/azblob/zc_appendBlobClient.go:77]: Yes that's correct. We can directly inherit the method from parent. I wonder why we have those methods in track 1.5 too

**Pull request merged**